stacker/providers/aws/default.py

```diff
@@ -239,7 +271,32 @@ def update_stack(self, fqn, template_url, old_parameters, parameters,
                     fqn,
                 )
                 raise exceptions.StackDidNotChange
-            raise
+            elif e.response['Error']['Message'] == ('TemplateURL must '
+                                                    'reference a valid '
+                                                    'S3 object to which '
+                                                    'you have access.'):
+                logger.warn("DEPRECATION WARNING: Falling back to legacy "
```

Comment:
Can you turn this logic/warning into a method, that way we don't duplicate quite so much code between create & update?